great_expectations/datasource/new_new_new_datasource.py

```diff
@@ -0,0 +1,67 @@
+#!!! Rename this
+from typing import Dict, List
+from great_expectations.core.batch import Batch
+from great_expectations.datasource.base_data_asset import BaseDataAsset
+from great_expectations.datasource.misc_types import NewBatchRequestBase
+from great_expectations.types import DictDot
+from great_expectations.validator.validator import Validator
+
+
+class NewNewNewDatasource:
+
+    def __init__(
+        self,
+        name,
+    ):
+        #!!! Trying this on for size
+        # experimental-v0.15.1
+        # warnings.warn(
+        #     "\n================================================================================\n" \
+        #     "NewNewNewDatasource is an experimental feature of Great Expectations\n" \
+        #     "You should consider the API to be unstable.\n" \
+        #     "\n" \
+        #     "You can disable this warning by calling: \n" \
+        #     "from great_expectations.warnings import GxExperimentalWarning\n" \
+        #     "warnings.simplefilter(action=\"ignore\", category=GxExperimentalWarning)\n" \
+        #     "\n" \
+        #     "If you have questions or feedback, please chime in at\n" \
+        #     "https://github.com/great-expectations/great_expectations/discussions/DISCUSSION-ID-GOES-HERE\n" \
+        #     "================================================================================\n",
+        #     GxExperimentalWarning,
+        # )
+
+        self._name = name
+        self._assets = DictDot()
+
+    def add_asset(self, name:str, *args, **kwargs) -> BaseDataAsset:
+        raise NotImplementedError
+
+    def rename_asset(self, old_name: str, new_name: str) -> None:
```

Comment:
IMHO this should be an asset operation, not a datasource one.

ex: person.name = "Gabe"

We can add magic to the assignment property to check for collisions

Reply:
Sure---the important thing is that the list of asset names on the Datasource must be kept in sync with the actual names of the DataAssets.

I don't have a strong opinion on which object is the driver for that.